server/src/test/kotlin/io/spine/server/procman/given/pm/QuizProcess.kt

```diff
@@ -0,0 +1,176 @@
+ * the question are answered using [Answer Question commands][PmAnswerQuestion].
+ */
+internal class QuizProcess(id: PmQuizId) : ProcessManager<PmQuizId, PmQuiz, PmQuiz.Builder>(id) {
+
+    @Assign
+    fun handle(command: PmStartQuiz): PmQuizStarted {
+        alter {
+            id = command.quiz
+            addAllOpenQuestion(command.questionList)
+        }
+        return pmQuizStarted {
+            quiz = command.quiz
+            question.addAll(command.questionList)
+        }
+    }
+
+    @Assign
+    fun handle(command: PmAnswerQuestion): PmQuestionAnswered =
+        pmQuestionAnswered {
+            quiz = command.quiz
+            answer = command.answer
+        }
+
+    @React
+    fun on(event: PmQuizStarted): Nothing {
+        alter { id = event.quiz }
+        return nothing()
+    }
+
+    @React
+    fun on(event: PmQuestionAnswered): EitherOf3<PmQuestionSolved, PmQuestionFailed, Nothing> {
+        val answer = event.answer
+        val question = answer.question
+        if (question.isClosed()) {
+            return EitherOf3.withC(nothing())
+        }
+        return if (answer.correct) {
+            EitherOf3.withA(pmQuestionSolved {
+                quiz = event.quiz
+                this.question = question
+            })
+        } else {
+            EitherOf3.withB(pmQuestionFailed {
+                quiz = event.quiz
+                this.question = question
+            })
+        }
+    }
+
+    private fun PmQuestionId.isClosed(): Boolean {
+        val openQuestions = builder().openQuestionList
+        val isOpen = openQuestions.contains(this)
+        return !isOpen
+    }
+
+    @React
+    fun on(event: PmQuestionSolved): EitherOf2<Nothing, PmQuizFinished> {
+        val question = event.question
+        alter {
+            removeOpenQuestion(question)
+            addSolvedQuestion(question)
+        }
+        return onAnsweredQuestion()
+    }
+
+    @React
+    fun on(event: PmQuestionFailed): EitherOf2<Nothing, PmQuizFinished> {
+        val question = event.question
+        alter {
+            removeOpenQuestion(question)
+            addFailedQuestion(question)
+        }
+        return onAnsweredQuestion()
+    }
+
+    private fun PmQuiz.Builder.removeOpenQuestion(question: PmQuestionId) {
+        val openQuestions = openQuestionList
+        val index = openQuestions.indexOf(question)
+        removeOpenQuestion(index)
+    }
+
+    private fun onAnsweredQuestion(): EitherOf2<Nothing, PmQuizFinished> {
+        return if (builder().openQuestionList.isEmpty()) {
+            val loaded = select(PmQuizStats::class.java).findById(id())
+            EitherOf2.withB(pmQuizFinished {
+                quiz = id()
+                loaded?.let {
+                    stats = it
+                }
+            })
+        } else {
+            EitherOf2.withA(nothing())
+        }
+    }
+}
+
+internal class QuizRepository : ProcessManagerRepository<PmQuizId, QuizProcess, PmQuiz>()
```

Comment:
Not sure why we need it. A default repo would do, with nothing customized in the behavior of this type.